internnav/agent/internvla_n1_agent.py

```diff
@@ -195,15 +220,15 @@ def should_infer_s2(self, mode="sync"):
                 return False
         # 2. Partial async mode: S2 infers 1 frame while S1 executes 10 frames
         if mode == "partial_async":
-            if self.episode_step - self.s2_output.idx >= 8:
+            if self.dual_forward_step >= self.sys2_max_forward_step:
                 return True
             if self.s2_output.output_action is None and self.s2_output.output_pixel is None and self.s2_output.output_latent is None:
                 # This normally only occurs when output is discrete action and discrete action has been fully executed
                 return True
             return False
-        # 3. Fully async mode: S2 and S1 run completely in parallel, so S2 infers every frame
-        if mode == "full_async":
-            return True
+        # # 3. Fully async mode: S2 and S1 run completely in parallel, so S2 infers every frame
```

Comment:
Fully async mode: S2 and S1 run completely in parallel, so S2 infers every frame. Currently, we do not support this mode.